### DISCUSSION TITLE
Understand "reduction_error_analysis" results

### DISCUSSION THREAD

**Discussion by CheLamVien** [pymor/pymor#2248] [Q&A] [answered]
Dear all, 

I am new to MOR. I have a question regarding the difference between "errors" (Array of the norms of the model reduction errors w.r.t. all given test Parameters, reduced basis dimensions and norms) and "error_estimates" (Array of the model reduction error estimates w.r.t. all given test Parameters and reduced basis dimensions.). I could not find the code for how to calculate "error_estimates". 

I ran the "thermalblock_simple.py" script and saw that the norm error is smaller than the error estimator, but I could not explain it. 
Any explanations are highly appreciated. 
Thank you so much for your time. 

Best regards, 
VC

**Comment by sdrave** [comment, marked as answer]
Hi @CheLamVien, I'm not sure if I understand your question correctly, but `error_estimates` contains a posteriori error estimates for the model reduction error contained in `errors`. Those estimates are always upper bounds and generally larger than the true error. (Good estimators have the property that the error is never overestimated by more than a fixed factor.) However, the crucial benefit of such an error estimator is that it can be computed from the reduced solution without knowing the actual full-order solution. To compute the true error, you to compute both the reduced-order and full-order solution.

In pyMOR, a posterior error estimates are computed using the `estimate_error` method of the reduced-order `Model`.

**Reply by CheLamVien** [reply]
Thank you so much for your explanation. I got it now. 
I have another question; I see an option for **NGsolve** binding in **PyMOR**. I have followed the [demo example ](https://github.com/pymor/pymor/blob/main/src/pymordemos/thermalblock_simple.py). I would like to know if I can use the iterative solver in **NGsolve**. 

Best regards, 
Vien Che

**Reply by sdrave** [reply]
Yes, you should be able to select the used linear solver by setting `solver_options` of `NGSolveMatrixOperator` or setting the `default_solver` `default` of `NGSolveMatrixOperator._prepare_apply`. If you need more customization, this could be easily added to the bindings.

**Reply by CheLamVien** [reply]
Thank you again. Now, I can use the CGSolver and preconditioner in **NGSolve**. I also would like to use the complex number. Could you please give me some hints?
I really appreciate any help you can provide.

**Reply by CheLamVien** [reply]
I think I have figured out how to use the complex number. I got the error: **raise ValueError(f'norm is complex (square = {norm_squared})')**. I changed **raise_complex=False**. Is this a correct solution, and why must we raise this value? What is the reason behind it? 
Thank you so much in advance.
